pkg/module/manager.go

```diff
@@ -0,0 +1,162 @@
+package module
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudquery/cloudquery/internal/file"
+	"github.com/cloudquery/cloudquery/modules"
+	"github.com/cloudquery/cloudquery/pkg/config"
+	"github.com/cloudquery/cloudquery/pkg/module/model"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/spf13/afero"
+)
+
+// ManagerImpl is the manager implementation struct.
+type ManagerImpl struct {
+	modules map[string]model.Module
+
+	// Instance of a database connection pool
+	pool *pgxpool.Pool
+
+	// Logger instance
+	logger hclog.Logger
+}
+
+// Manager is the interface that describes the interaction with the module manager.
+// Implemented by ManagerImpl.
+type Manager interface {
+	// RegisterModule is used to register a module into the manager.
+	RegisterModule(mod model.Module)
+
+	// ParseModuleReference parses and validates the given module name and config into an execution request.
+	ParseModuleReference(ctx context.Context, baseReq model.ExecuteRequest, modName string, modConfigPath string) (*model.ExecuteRequest, error)
+
+	// ExecuteModule executes the given module.
+	ExecuteModule(ctx context.Context, execRequest *model.ExecuteRequest) (*model.ExecutionResult, error)
+
+	// ReadConfig reads the given module's default/builtin config
+	ReadConfig(modName string) ([]byte, error)
+}
+
+// NewManager returns a new manager instance.
+func NewManager(pool *pgxpool.Pool, logger hclog.Logger) *ManagerImpl {
+	return &ManagerImpl{
+		modules: make(map[string]model.Module),
+		pool:    pool,
+		logger:  logger,
+	}
+}
+
+// RegisterModule is used to register a module into the manager.
+func (m *ManagerImpl) RegisterModule(mod model.Module) {
+	m.modules[mod.ID()] = mod
+}
+
+// ParseModuleReference parses and validates the given module name and config into an execution request.
+func (m *ManagerImpl) ParseModuleReference(ctx context.Context, baseReq model.ExecuteRequest, modName string, modConfigPath string) (*model.ExecuteRequest, error) {
+	mod, ok := m.modules[modName]
+	if !ok {
+		return nil, fmt.Errorf("module not found %q", modName)
+	}
+
+	rawConfig, err := m.readDecodeConfig(mod.ID(), modConfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("could not read config: %w", err)
+	}
+
+	if err := mod.Configure(ctx, rawConfig); err != nil {
+		return nil, fmt.Errorf("module configuration failed: %w", err)
+	}
+
+	baseReq.Module = mod
+
+	return &baseReq, nil
+}
+
+// ExecuteModule executes the given module.
+func (m *ManagerImpl) ExecuteModule(ctx context.Context, execReq *model.ExecuteRequest) (*model.ExecutionResult, error) {
+	// Acquire connection from the connection pool
+	theConn, err := m.pool.Acquire(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to acquire connection from the connection pool: %w", err)
+	}
+	defer theConn.Release()
+
+	execReq.Conn = theConn
+
+	res := execReq.Module.Execute(ctx, execReq)
```

Comment:
this is the module's own execute method. adding arbitrary args here would break compatibility, better contain them in a single exec request. executerequest.module is only used in the stringer, let me remove that so that it's clear.